server/models/image_item.py

```diff
@@ -256,6 +258,8 @@ def getThumbnail(self, item, width=None, height=None, **kwargs):
         """
         # check if a thumbnail file exists with a particular key
         keydict = dict(kwargs, width=width, height=height)
+        for key in [key for key in keydict if keydict[key] is None]:
```

Comment:
A dict comprehension would probably be more readable here.

    keydict = {k: v for k, v in six.viewitems(keydict) if v is not None}